### ISSUE TITLE
Implement TextFormField

### ISSUE THREAD

**Issue by grahamsmith** [jaumard/sms_autofill#82]
When using this library in forms it would be really useful to allow TextFormField with a validator on it for PhoneFieldHint. I am happy to implement this however I want some direction on approach.

There are a few ways this could be done:

- Brand new PhoneFormFieldHint, which would be 99% the same as PhoneFieldHint except it takes a validator.
- An option to state that a TextFormField should be used over TextField when PhoneFieldHint is created.

```
@override
  Widget build(BuildContext context) {
    return widget.child ??
        TextField(
          autofocus: widget.autofocus,
          focusNode: _focusNode,
          autofillHints: [AutofillHints.telephoneNumber],
          inputFormatters: _inputFormatters,
          decoration: _decoration,
          controller: _controller,
          keyboardType: TextInputType.phone,
        );
  }
```

could become

```
@override
  Widget build(BuildContext context) {
    return widget.child ?? createField(isFormWidget);
  }
  
  Widget createField(bool isFormWidget) {
    if(isFormWidget) _createTextFormField(); else _createTextField();
  }

  Widget _createTextField() {
    return TextField(
      autofocus: widget.autofocus,
      focusNode: _focusNode,
      autofillHints: [AutofillHints.telephoneNumber],
      inputFormatters: _inputFormatters,
      decoration: _decoration,
      controller: _controller,
      keyboardType: TextInputType.phone,
    );
  }

  Widget _createTextFormField() {
    return TextFormField(
      validator: validator, // we can have validators :)
      autofocus: widget.autofocus,
      focusNode: _focusNode,
      autofillHints: [AutofillHints.telephoneNumber],
      inputFormatters: _inputFormatters,
      decoration: _decoration,
      controller: _controller,
      keyboardType: TextInputType.phone,
    );
```

@jaumard some thoughts would be great on what would be prefered.

**Comment by grahamsmith** [comment]
This is now resolved.

@jaumard - 🙏  can we get a new release out when you get time please? I really could do with the enhancements made.

**Comment by jaumard** [comment]
sure  :) fixed on 1.3.0 thanks for your contributions!
